packages/protocol/scripts/bash/get_smart_contract_size_from_build_artifacts.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+: '
+  Uses:
+    1. Truffle build artifacts located in the packages/protocol/build/ directory
+
+  Requirements:
+  1. Have the Truffle build artifacts in the packages/protocol/build/ directory
+  '
+
+# Get the current date and time for the filename
+TIMESTAMP=$(date +"%Y%m%d_%H%M%S")
+
+# Get the directory where the script is located, assuming the script is in the root or a subdirectory of the repo
+BASE_DIRECTORY="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
+# Get the root of the repo based on the known structure
+REPO_ROOT_DIRECTORY="$BASE_DIRECTORY/../../../"
+
+# Get the directory with build artifacts
+BUILD_ARTIFACT_DIRECTORY="${REPO_ROOT_DIRECTORY}protocol/build/contracts"
+
+# Create the output directory if it doesn't exist
+OUTPUT_DIRECTORY="$BASE_DIRECTORY/out"
+mkdir -p "$OUTPUT_DIRECTORY"
+OUTPUT_FILE="$OUTPUT_DIRECTORY/build_artefact_bytecode_sizes_$TIMESTAMP.csv"
+
+# Temporary file to collect data before sorting
+TEMP_FILE=$(mktemp)
+
+# Write the header to the output file
+echo "Contract,Size (KB)" > "$OUTPUT_FILE"
+
+# Find all JSON files in the subdirectories and process them
+find "$BUILD_ARTIFACT_DIRECTORY" \
+     "${BUILD_ARTIFACT_DIRECTORY}-0.8" \
+     "${BUILD_ARTIFACT_DIRECTORY}-mento" -name '*.json' | while read -r file
+do
+    # Extract the contract name and bytecode using jq
+    contract=$(jq -r '.contractName' "$file")
+    bytecode=$(jq -r '.bytecode' "$file")
```

Comment:
That's a great point, [EIP-170](https://eips.ethereum.org/EIPS/eip-170) puts "a hard cap on the size of an object that can be **saved to the blockchain**".

Since it's the `deployedBytecode` that will be stored in the smart contract, and not the initialization `bytecode`, we should be looking at the size of the `deployedBytecode`.

That might also explain (parts of) why the on-chain calculation is slightly different, since that uses the `eth_getCode` RPC method, which returns the deployed bytecode:

> The command `eth_getCode` returns the bytecode associated with the address in the Ethereum world state, that will be the deployed bytecode (or runtime bytecode).
> 
> The init bytecode could be obtained from the deployment transaction's data field, unless it was created by a contract in which case you have to trace the transaction to obtain the details.

Source: [ethereum.stackexchange.com](https://ethereum.stackexchange.com/questions/108152/does-eth-getcode-return-the-init-bytecode-or-the-deployed-bytecode)

Let me update this 👍

Reply:
That's done in https://github.com/celo-org/celo-monorepo/pull/10989/commits/c128c7f1b417a8cd82fab2d93c84c4ea00b50678. 

With this change the contract sizes using on-chain method and the build artifact method are equal.

<img width="1039" alt="image" src="https://github.com/celo-org/celo-monorepo/assets/46296830/7f412ad5-4378-441b-99fa-84bf4251f4e6">

Source: [Contract sizes](https://docs.google.com/spreadsheets/d/1MPo5mtC8DT8d3y1ovGGhj04rQ3Zx6Xb0prgz7F4VNQM/edit#gid=1494083158)